play.pokemonshowdown.com/js/storage.js

```diff
@@ -913,10 +918,18 @@ Storage.fastUnpackTeam = function (buf) {
 		i = j + 1;
 
 		// moves
-		j = buf.indexOf('|', i);
+		j = buf.indexOf(';', i);
+		if (j < 0) j = buf.indexOf('|', i);
 		set.moves = buf.substring(i, j).split(',');
 		i = j + 1;
 
+		// move PP ups
+		if (buf.charAt(j) === ';') {
+			j = buf.indexOf('|', i);
+			set.movePPUps = buf.substring(i, j).split(',').map(number => parseInt(number));
```

Comment:
```suggestion
			for (var q = 0; q < buf.substring(i, j).split(',').length; q++) {
				var ppUpsUsed = parseInt(buf.substring(i, j).split(',')[q]);
				set.movePPUps[q] = ppUpsUsed;
			}
```

Reply:
You might want to pull `buf.substring(i,j).split(,)` into a variable outside the for loop. Also `set.movePPUps` needs to be initialized w/ an array before writing to it.

Reply:
Initializing the array with the values as strings and converting them to numbers works without crashing or upsetting Linter, so I've done that.